### ISSUE TITLE
could not reproduce the claimed result

### ISSUE THREAD

**Issue by CircleAAA** [MIT-SPARK/MiDiffusion#2]
As pretrained weights are not provided so far, I downloaded 3D-FRONT and trained the model following your instructions on both bedroom and livingroom dataset. I changed the feature_extractor in the config file from pointnet_simple to resnet18 as below. Besides, I only changed some paths and dirs.
```
#feature_extractor:
#    name: "pointnet_simple"
#    feat_units: [4, 64, 64, 512, 64]

feature_extractor:
    name: "resnet18"
    feature_size: 64
    freeze_bn: true
    input_channels: 1
```
The loss for livingroom finally converges at around 0.039. Though I adjusted the learning rate to 0.00001, there is no substantial reduce in the loss even after hundreds of epochs. So is the loss for bedroom, which finally oscillated about 0.028.

**I rendered the test sample and they did not look good quality.** 
Can you explain this or figure out if something I did wrong? 
Some of the test results are shown below.

![0](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/6ce5eb02-e47c-4f10-ab35-7d4d12def622)
![1](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/4ce6702c-725a-4482-a2d5-6d130efa6b17)
![2](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/5d4153a1-5862-4c95-80ff-6667764d74ab)
![3](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/72529df3-a29f-47d4-846b-7a595a1e4c64)
![4](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/95f75fb7-7768-41e0-bd92-760580ba4b23)
![5](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/4d20ff46-1793-4716-b3c0-96a374d40303)
![6](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/135fd7a9-f33b-4231-a7cd-347a167b7ae0)
![7](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/c90bae63-4224-4dc1-8a3f-e596906b175b)
![8](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/ffa31916-f0e7-4a92-b9a5-79f369712895)
![9](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/cae58dda-dff2-4ea0-bcac-2dc54838ee09)
![10](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/a60e6751-51de-4a2b-a06a-15cba7842fd7)
![11](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/c41a2e3c-8b60-4d4b-b26e-98719f612864)
![12](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/6c0ca050-126d-4c2d-a033-fd676ebc9b92)
![13](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/f6aed7bf-9dff-4333-b6ad-353917761364)
![14](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/f1059191-ff94-474c-b7a8-b77199259eac)
![15](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/543df25f-ab07-47b6-b748-b65ee99fdf43)

**Comment by SiyiHu** [comment]
Your modification to the config file is the correct way to switch from PointNet to ResNet-18 floor plan feature extractor. We have not tried training rate that low. We also did not do much parameter tuning for models using ResNet-18, since this is not our key contribution. Based on experience training DiffuScene models (which uses ResNet-18) and ours, I would guess you need at least 60-65k epochs (if you keep all other parameters in the config file). You can also try 100k epochs. The validation loss overfit might be pretty big, but the final model should be able to generate reasonable layouts.

**Comment by CircleAAA** [comment]
The loss went down to around 0.011 for bedroom scenes after 18k epochs. The evaluation results on testset are very close to that claimed in paper. While, some pieces of furniture are not correctly oriented. Do you recomment me train more epochs or revise the loss function?

![1](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/13b69b4f-8f27-410c-b485-52d36f3dd399)
![2](https://github.com/MIT-SPARK/MiDiffusion/assets/145274399/e506096c-bae9-4bb0-815a-2354b254f2c9)

**Comment by SiyiHu** [comment]
I have not run evaluations around 18k epochs. For bedroom, I am pretty sure there are non-trivial improvements from 30k to 50k. If furniture is consistently off by 180 deg (which does happen to some of our runs), you can try removing "cosin_angle" in the config file.

**Comment by CircleAAA** [comment]
The loss for bedroom scenes went down to around 0.0088 after 45k epochs. The orientation problem remains, and I will try removing cosine angle then. For livingrooms, I trained 100k epochs and the performance on testset looks good.

**Comment by foojayyy** [comment]
> The loss for bedroom scenes went down to around 0.0088 after 45k epochs. The orientation problem remains, and I will try removing cosine angle then. For livingrooms, I trained 100k epochs and the performance on testset looks good.
Did you solve the orientation problem? I encountered it too.Thank you.

**Comment by foojayyy** [comment]
> The loss for bedroom scenes went down to around 0.0088 after 45k epochs. The orientation problem remains, and I will try removing cosine angle then. For livingrooms, I trained 100k epochs and the performance on testset looks good.





Did you solve the orientation problem? I encountered it too.

**Comment by orpheus92** [comment]
I am using the pre-trained model on custom dataset. However, I also encountered orientation problem. Anyone knows how to solve that?
